### ISSUE TITLE
Question: why when I run `slate watch` i get redirected to "store.com/password" instead of local dev environment?

### ISSUE THREAD

**Issue by rublev** [Shopify/slate#182]
### Problem

When I run `slate watch` on a theme, everything generates just fine in the console and the local dev url launches, except it redirects to my live store password page. Is there a common reason for this? Unsure what kind of code to include as this is happening with multiple themes. I am currently editing my live theme.

### Video

![wat-shopify-slate](https://user-images.githubusercontent.com/470273/27512392-4a92393c-590c-11e7-9195-f1361cbbf485.gif)

**Comment by kyl-newton** [comment]
Hi @rublev. I am experiencing this issue and have found this issue occurs when both:

- Store password is enabled
- Custom domain has been added and traffic is being redirected to it

The short term solution for me was to set the primary domain back to the myshop.shopify.com domain which makes the local environment work again which is fine for development is the store has not launched (I guess).

I think it also works if you disable the password page and set the domain back to your custom domain which would also be acceptable for live development.

However, I'm sure there will be some cases where developers need to update a live site with a custom domain that is password protected and that is where this issue is, in fact, a bug that should be fixed.

**Comment by rublev** [comment]
Thanks for the input @kyl-newton, much appreciated. I agree, it'd be nice if this got fixed. Could .htpasswd or add a custom PW page to your theme.

**Comment by paulmist** [comment]
I also have this issue but, only when attempting to access the browsersync external URL.

**Comment by NathanPJF** [comment]
I've made an issue in the Slate tools repo outlining the problems seen with Browsersync and local host.

@kyl-newton Thank you for pointing out that having a password protected storefront with a custom domain is at the root.

As a temporary work around: in testing I've found that if I don't enable a forced redirect to the primary domain and then manually correct the `action` parameter on the password form, I can continue working on the password protected site.  I go into more detail on the linked issue:

> The issue here is the `action` parameter. With the correct action attribute, the site works as expected. Video of me changing the action manually in the inspector: https://screenshot.click/screencast_2017-07-09_21-52-23.mp4

^ Definitely not a solution, but maybe it can help some people with this issue until a fixed is pushed.
